File: Udemy/Practicas/CursoDePython_Udemy/Calculadora.py
# Suma
def Suma(cant):
    suma = None
    print()

    for i in range(cant):
        num = eval(input(f"Digita el número {i + 1}: "))
        if suma == None:
            suma = num
        else:
            suma = suma + num
        
    print(f"\nEl resultado de la suma es: {suma}")
    return(suma)

# Resta
def Resta(cant):
    resta = None
    print()

    for i in range(cant):
        num = eval(input(f"Digita el número {i + 1}: "))
        if resta == None:
            resta = num
        else:
            resta = resta - num

    print(f"\nEl resultado de la resta es: {resta}")
    return(resta)

# Multiplicacion
def Multiplicacion(cant):
    multiplicacion = None
    print()

    for i in range(cant):
        num = eval(input(f"Digita el número {i + 1}: "))
        if multiplicacion == None:
            multiplicacion = num
        else:
            multiplicacion = multiplicacion * num

    print(f"\nEl resultado de la multiplicacion es: {multiplicacion}")
    return(multiplicacion)

# Division
def Division(cant):
    division = None
    print()

    for i in range(cant):
        num = eval(input(f"Digita el número {i + 1}: "))
        if division == None:
            division = num
        else:
            division = division / num

    print(f"\nEl resultado de la division es: {division}")
    return(division)

# Modulo de una división
def Modulo(cant):
    modulo = None
    print()

    for i in range(cant):
        num = eval(input(f"Digita el número {i + 1}: "))
        if modulo == None:
            modulo = num
        else:
            modulo = modulo % num

    print(f"\nEl resultado de la modulo de la división es: {modulo}")
    return(modulo)

def menu(nombre):
    opcion = eval(input(f"\nQué deseas hacer {nombre}?: \n"
                         "\n1.) Sumar."
                         "\n2.) Restar."
                         "\n3.) Multiplicar"
                         "\n4.) Dividir"
                         "\n5.) Modulo de un división."
                         "\n0.) Salir \n"
                         "\nOpción: "))
    return(opcion)

def main():
    nombre = input("\nBienvenido..." +
                   "\nDigita tu nombre: ")
    opcion = menu(nombre)

    while(True):
        # Suma
        if (opcion == 1):
            print("\nHaz elegido la opción \"1.) Sumar.\"")
            cantidad = eval(input("\nDigita una cantidad de números a sumar: "))
            Suma(cantidad)

        # Resta
        elif (opcion == 2):
            print("\nHaz elegido la opción \"2.) Restar.\"")
            cantidad = eval(input("\nDigita una cantidad de números a restar: "))
            Resta(cantidad)

        # Multiplicación
        elif (opcion == 3):
            print("\nHaz elegido la opción \"3.) Multiplicar.\"")
            cantidad = eval(input("\nDigita una cantidad de números a multiplicar: "))
            Multiplicacion(cantidad)

        # División
        elif (opcion == 4):
            print("\nHaz elegido la opción \"3.) Dividir.\"")
            cantidad = eval(input("\nDigita una cantidad de números a dividir: "))
            Division(cantidad)

        # Módulo
        elif (opcion == 5):
            print("\nHaz elegido la opción \"3.) Modulo de una división.\"" +
                  "\nSolo debes digitar 2 números: ")
            cantidad = 2
            Modulo(cantidad)

        # Salir
        elif (opcion == 0):
            print("\nHaz elegido la opcion \"0.) Salir\"... \n" +
                  "\nHasta pronto!!!... \n")
            break

        # Error
        else:
            print("\nError!!!... Opción incorrecta!!!")

        opcion = menu(nombre)

main()

File: Udemy/Practicas/Curso_De_C#_Udemy/Calculadora3/Calculadora.cs
// Clase para operaciones matemáticas
namespace Calculadora3
{
    interface Calculadora
    {
        public int Menu(string nombre);
        public double? Suma(int cant);
        public double? Resta(int cant);
        public double? Multiplicacion(int cant);
        public double? Division(int cant);
        public double? Modulo(int cant);
    }

    public class  Operaciones : Calculadora
    {
        // Función que muestra un menú de opciones
        public int Menu(string nombre)
        {
            int opcion;

            Console.Write($"\nQué deseas hacer {nombre}?: \n" +
                           "\n1.) Sumar." +
                           "\n2.) Restar." +
                           "\n3.) Multiplicar" +
                           "\n4.) Dividir" +
                           "\n5.) Módulo de un división." +
                           "\n0.) Salir \n" +
                           "\nOpción: ");

            opcion = int.Parse(Console.ReadLine()!);
            return opcion;
        }

        // Función que calcula el resultado de una suma
        public double? Suma(int cant)
        {
            double? suma = null;
            double numero;

            for (var i = 0; i < cant; i++)
            {
                Console.Write($"Digita el número {i + 1}: ");
                numero = double.Parse(Console.ReadLine()!);

                if (suma is null)
                {
                    suma = numero;
                } else
                {
                    suma = suma + numero;
                }
            }

            Console.WriteLine($"\nEl resultado de la suma es: {suma}");
            return suma;
        }

        // Función que calcula el resultado de una resta
        public double? Resta(int cant)
        {
            double? resta = null;
            double numero;

            for (var i = 0; i < cant; i++)
            {
                Console.Write($"Digita el número {i + 1}: ");
                numero = double.Parse(Console.ReadLine()!);

                if (resta is null)
                {
                    resta = numero;
                } else
                {
                    resta = resta - numero;
                }
            }

            Console.WriteLine($"\nEl resultado de la resta es: {resta}");
            return resta;
        }

        // Función que calcula el resultado de una multiplicación
        public double? Multiplicacion(int cant)
        {
            double? multiplicacion = null;
            double numero;

            for (var i = 0; i < cant; i++)
            {
                Console.Write($"Digita el número {i + 1}: ");
                numero = double.Parse(Console.ReadLine()!);

                if (multiplicacion is null)
                {
                    multiplicacion = numero;
                } else
                {
                    multiplicacion = multiplicacion * numero;
                }
            }

            Console.WriteLine($"\nEl resultado de la multiplicación es: {multiplicacion}");
            return multiplicacion;
        }

        // Función que calcula el resultado de una devisión
        public double? Division(int cant)
        {
            double? division = null;
            double numero;

            
            for (var i = 0; i < cant; i++)
            {
                Console.Write($"Digita el número {i + 1}: ");
                numero = double.Parse(Console.ReadLine()!);

                if (division is null)
                {
                    division = numero;
                } else
                {
                    division = division / numero;
                }
            }

            Console.WriteLine($"\nEl resultado de la división es: {division}");
            return division;
        }

        // Función que calcula el resultado del módulo de una división
        public double? Modulo(int cant)
        {
            double? modulo = null;
            double numero;

            for (var i = 0; i < cant; i++)
            {
                Console.Write($"Digita el número {i + 1}: ");
                numero = double.Parse(Console.ReadLine()!);
                
                if (modulo is null)
                {
                    modulo = numero;
                } else
                {
                    modulo = modulo % numero;
                }
            }

            Console.WriteLine($"\nEl resultado de la división es: {modulo}");
            return modulo;
        }
    }
}
